frontend/packages/metal3-plugin/src/components/host.tsx

```diff
@@ -208,9 +200,34 @@ export const BaremetalHostsPage: React.FC<BaremetalHostsPageProps> = (props) =>
       rowFilters={[hostStatusFilter]}
       createButtonText="Add Host"
       namespace={props.namespace}
-      resources={[hostsResource, machinesResource, nodesResource]}
+      resources={resources}
       flatten={flatten}
       ListComponent={HostList}
+      customData={{
+        hasNodeMaintenanceCapability: props.hasNodeMaintenanceCapability,
+      }}
     />
   );
 };
+
+const hostsPageStateToProps = ({ k8s }) => ({
+  hasNodeMaintenanceCapability: !!k8s.getIn([
+    'RESOURCES',
+    'models',
+    referenceForModel(NodeMaintenanceModel),
+  ]),
+});
+
+export const BaremetalHostsPageConnected = connect(hostsPageStateToProps)(BaremetalHostsPage);
+
+type BaremetalHostsPageProps = {
```

Comment:
I'd prefer to colocate the prop types with the component rather than putting it to the bottom of the file. It forces me to jump up and down in the file to check the props.

Reply:
Existing Console convention is to list all types (in the same file) at the bottom. Plugins should have some degree of freedom as to how their code is organized, so I don't see any issue with co-location of type declarations, if the package maintainers prefer it that way.

Reply:
reverted